packages/core/src/integrations/zoderrors.ts

```diff
@@ -56,64 +77,142 @@ function formatIssueTitle(issue: ZodIssue): SingleLevelZodIssue<ZodIssue> {
   };
 }
 
+/**
+ * Takes ZodError issue path array and returns a flattened version as a string.
+ * This makes it easier to display paths within a Sentry error message.
+ *
+ * Array indexes are normalized to reduce duplicate entries
```

Comment:
I'm not sure we want to duplicate entries here, I think `[0].foo.[1].bar` is more valuable than `<array>.foo.<array>.bar`, and we want to make sure that different array entries will lead to different issues.

What do you think?

Reply:
This is only applied to the issue message, not `event.extra` or attachments.

In practice, we found that having the indexes in the message is more noise than helpful for understanding the issue at a glance, which was the primary motivation for me making this improvement in our internal fork.

Reply:
Sounds good to me, let's keep what is here then